apps/cowswap-frontend/src/legacy/hooks/useRecentActivity.ts

```diff
@@ -163,9 +185,9 @@ export function createActivityDescriptor(tx?: EnhancedTransactionDetails, order?
     id = tx.hash
 
     const isReceiptConfirmed =
-      tx.receipt?.status === TxReceiptStatus.CONFIRMED || typeof tx.receipt?.status === 'undefined'
+      !!tx.receipt && (tx.receipt.status === TxReceiptStatus.CONFIRMED || typeof tx.receipt.status === 'undefined')
```

Comment:
Fixed the wrong condition.
When there is no `tx.receipt` then `isReceiptConfirmed` will be true, because tx.receipt?.status === 'undefined' is true